File: Algorithm_605_Can Place Flowers.cpp
#include <vector>
using namespace std;
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int len=flowerbed.size();
        if (n>len) return false;
        if (flowerbed[0]==0 && flowerbed[1]==0)
        {
            n--;
            flowerbed[0]=1;
        }
        if (flowerbed[len-1]==0 && flowerbed[len-2]==0)
        {
            n--;
            flowerbed[len-1]=1;
        }
        for (int i=1;i<len-1;i++)
        {
            if (flowerbed[i-1] == 0 && flowerbed[i+1] == 0 && flowerbed[i] == 0)
            {
                n--;
                flowerbed[i]=1;
            }
        }
        return n<=0;
    }
};

File: Algorithm_605_Can Place Flowers.java
// 思路：贪心法，只要遇到能够种花的位置，就种花上去

class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int count = 0;
        for (int i = 0; i < flowerbed.length; i++) {
            if (isValid(flowerbed, i)) {
                flowerbed[i] = 1;
                count++;
            }
        }
        return count >= n;
    }

    private boolean isValid(int[] flowerbed, int i) {
        boolean leftValid = i == 0 || flowerbed[i - 1] == 0;
        boolean rightValid = i == flowerbed.length - 1 || flowerbed[i + 1] == 0;
        return flowerbed[i] == 0 && leftValid && rightValid;
    }
}